bokehjs/src/coffee/renderer/tile/mercator_tile_source.coffee

```diff
@@ -0,0 +1,241 @@
+_ = require "underscore"
+TileSource = require "./tile_source"
+
+class MercatorTileSource extends TileSource
+
+  type: 'MercatorTileSource'
+  
+  defaults: =>
+    return _.extend {}, super(), {
+      tile_size : 256
+      max_zoom : 30
+      min_zoom : 0
+      extra_url_vars : {}
+      full_extent : [-20037508.34, -20037508.34, 20037508.34, 20037508.34]
+      x_origin_offset : 20037508.34
+      y_origin_offset : 20037508.34
+      initial_resolution : 2 * Math.PI * 6378137 / 256
+      resolutions : undefined
```

Comment:
would it not be better to use proj4 calculations for the extents and offset and resolution - it may make it clearer why this random collection of numbers.